.github/workflows/test-pr.yml

```diff
@@ -256,9 +269,16 @@ jobs:
   test-nix:
     name: 'Nix'
     strategy:
+      fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-11]
-    runs-on: ${{ matrix.os }}
+        include:
+          - runner: ubuntu-latest
+            os: ubuntu-latest
+          - runner: macos-12
+            os: macos-12
+          - runner: MacM1
+            os: self-macos-12
```

Comment:
I'm not sure the old-style nix expression will work on the M1 mac. Seems its failing on a seg-fault atm, so could try adding `GC_DONT_GC=1` to the `nix-build` commands. Otherwise best to just remove, as we will be migrating away from these old nix derivations at some point anyway.

Reply:
If the long-term intent is to deprecate those derivations, I'll probably just remove it from this matrix. For this platform we're already being pretty aggressive about where you can build things due to tool compatibility, so having the flake build be the one true build shouldn't be an issue.